### DISCUSSION TITLE
Unexpected Error When Mentioning Assistants (@) – TypeError: Cannot Read Property 'map' of Undefined

### DISCUSSION THREAD

**Discussion by hanna-daoud** [danny-avila/LibreChat#4112] [Troubleshooting]
### What happened?

When I attempt to mention assistants in the chat by typing @ and selecting one, I receive an unexpected error in the application. The error prevents me from proceeding with the assistant selection and crashes the input field.

**Expected Behavior:**
I expected to see a list of available assistants and to be able to select one without the application crashing.

**Environment:**
Branch: Main
Operating System: Ubuntu 24.04
Browser: Chrome Version 128.0.6613.137 (Official Build) (64-bit)
Node.js Version: v20.17.0

### Steps to Reproduce

1. Open LibreChat.
2. Start a new chat and type @ in the input field.
3. Try to select any assistant from the dropdown menu.
4. The application crashes, and an error is displayed.

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
Unexpected Application Error!
Cannot read properties of undefined (reading 'map')
TypeError: Cannot read properties of undefined (reading 'map')
    at Mention (Mention.tsx:159:43)
    at renderWithHooks (chunk-AXVHJUCK.js:21682:26)
    ...

I am running the application in debug mode, and the error occurs in the Mention.tsx file on line 159. Here's the relevant part of the code:

{(matches as MentionOption[]).map((mention, index) => (
```

### Screenshots

![image](https://github.com/user-attachments/assets/5a7f39b9-1337-40e3-a166-2d6fd9ea6cae)
![image](https://github.com/user-attachments/assets/b35ceb11-ed1e-42a8-8b71-77edc1446918)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
### Can't reproduce this:


https://github.com/user-attachments/assets/16b277e5-9118-4158-901c-7efc43c55fdb

### Some followup questions:

- do you have any assistants created already?
- have you used this successfully before?
    - if so, when did this start?
- This used to be an issue and was fixed here: https://github.com/danny-avila/LibreChat/pull/2966
    - is your instance forked and/or do you have the fix from upstream?

**Reply by hanna-daoud** [reply]
Apparently, after properly setting my OpenAI Assistants API Key, no issue is being faced. But if I do not have an OpenAI Assistants API Key setup, the bug happens.

**Reply by danny-avila** [reply]
Thanks, I will try that, likely an edge case there
